rasa/core/training/story_conflict.py

```diff
@@ -146,56 +155,94 @@ def find_story_conflicts(
     trackers: List[TrackerWithCachedStates],
     domain: Domain,
     max_history: Optional[int] = None,
+    nlu_config: Optional[RasaNLUModelConfig] = None,
 ) -> List[StoryConflict]:
     """Generates `StoryConflict` objects, describing conflicts in the given trackers.
 
     Args:
         trackers: Trackers in which to search for conflicts.
         domain: The domain.
         max_history: The maximum history length to be taken into account.
+        nlu_config: NLU config.
 
     Returns:
         StoryConflict objects.
     """
-    if not max_history:
-        max_history = _get_length_of_longest_story(trackers, domain)
+    if max_history:
+        logger.info(
+            f"Considering the preceding {max_history} turns for conflict analysis."
+        )
+    else:
+        logger.info("Considering all preceding turns for conflict analysis.")
 
-    logger.info(f"Considering the preceding {max_history} turns for conflict analysis.")
+    tokenizer = _get_tokenizer_from_nlu_config(nlu_config)
 
     # We do this in two steps, to reduce memory consumption:
 
     # Create a 'state -> list of actions' dict, where the state is
     # represented by its hash
     conflicting_state_action_mapping = _find_conflicting_states(
-        trackers, domain, max_history
+        trackers, domain, max_history, tokenizer
     )
 
     # Iterate once more over all states and note the (unhashed) state,
     # for which a conflict occurs
     conflicts = _build_conflicts_from_states(
-        trackers, domain, max_history, conflicting_state_action_mapping
+        trackers, domain, max_history, conflicting_state_action_mapping, tokenizer,
     )
 
     return conflicts
 
 
+def _get_tokenizer_from_nlu_config(
+    nlu_config: Optional[RasaNLUModelConfig] = None,
+) -> Optional[Tokenizer]:
+    """Extracts the `tokenize` function of the first Tokenizer in the pipeline.
+
+    Args:
+        nlu_config: NLU Config.
+    """
+    if not nlu_config:
+        return None
+
+    pipeline: List[Component] = Trainer(
+        nlu_config, skip_validation=True
+    ).pipeline  # ToDo: ComponentBuilder?
+    tokenizer: Optional[Tokenizer] = None
+    for component in pipeline:
+        if isinstance(component, Tokenizer) and tokenizer:
```

Comment:
As you'll only add tokenizers anyway
```suggestion
        if tokenizer:
```